### PR TITLE
chore: Add Redis password option to helm chart

### LINKED ISSUE #13995: Allow password to be specified for redis
**Is your feature request related to a problem? Please describe.**
https://github.com/apache/superset/blob/master/helm/superset/values.yaml says you can set a password for redis. (search for `Redis password`)
However, the superset helm chart does not appear to accept a redis password configuration.
See https://github.com/apache/superset/blob/master/helm/superset/templates/_helpers.tpl#L60 and 
https://github.com/apache/superset/blob/master/helm/superset/templates/secret-env.yaml

**Describe the solution you'd like**
Helm chart modified to allow passwords.

**Describe alternatives you've considered**
Redis can be locked down in a VPC. This doesn't completely eliminate the need for passwords but it helps protect it.

**Additional context**
Related issue: https://github.com/apache/superset/issues/13541

### PR DESCRIPTION
**Pull request by robdiciuccio**
### SUMMARY
Adds the ability to set Redis password and DB values in the Helm chart.

### TEST PLAN
Config values tested locally. Deploy the updated Helm chart to confirm correct operation.

### ADDITIONAL INFORMATION

- [x] Has associated issue: Fixes #13995
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: helm/superset/templates/_helpers.tpl [hunk 1]
```diff
@@ -63,8 +63,8 @@ CACHE_CONFIG = {
       'CACHE_KEY_PREFIX': 'superset_',
       'CACHE_REDIS_HOST': env('REDIS_HOST'),
       'CACHE_REDIS_PORT': env('REDIS_PORT'),
-      'CACHE_REDIS_DB': 1,
-      'CACHE_REDIS_URL': f"redis://{env('REDIS_HOST')}:{env('REDIS_PORT')}/1"
```
Comment (dpgaspar) [inline helm/superset/templates/_helpers.tpl:67]: isn't `CACHE_REDIS_URL` needed?

Comment (robdiciuccio) [reply helm/superset/templates/_helpers.tpl:67]: Not if the other options are preset (it's essentially duplicating the above). Tested this config locally, and works without `CACHE_REDIS_URL`.

**Review by craig-rueda — APPROVED** [review]

**Review by dpgaspar — APPROVED** [review]

**Comment by julianbei** [comment]
Sorry but didnt you miss the celery config in setting the password?

**Comment by julianbei** [comment]
Here: https://github.com/apache/superset/pull/14274/files#diff-7ec63d52f04a74d3ad45aff0ad20600bfbc42c5fa219d84d4495c696542054c4R91

Add: 
``` 
password=env('REDIS_PASSWORD'),
```

**Comment by caleb15** [comment]
@robdiciuccio just following up on the above to see if you saw it. If you did and are busy no worries. We might have time to fix it at some point.

**Pull request merged**